pkg/jaeger/store/translation.go

```diff
@@ -44,8 +45,30 @@ var (
 		"producer":    "producer",
 		"unspecified": "",
 	}
+	// Our tracing implementation is based on OTLP spec, though OTLP spec
+	// doesn't mandate span name to be non empty[1], we have a constraint
+	// in the _ps_trace.operation table which raises error on empty span name.
+	// On the other hand, Jaeger span name can be empty and their storage
+	// integration tests validates the same[2]. Hence we replace empty span name
+	// with unspecifiedOperationName to make database happy.
+	//
+	// [1] https://github.com/open-telemetry/opentelemetry-proto/blob/724e427879e3d2bae2edc0218fff06e37b9eb46e/opentelemetry/proto/trace/v1/trace.proto#L110-L121
+	// [2] https://github.com/jaegertracing/jaeger/blob/7872d1b07439c3f2d316065b1fd53e885b26a66f/plugin/storage/integration/fixtures/traces/default.json#L6
+	unspecifiedOperationName = "unspecified"
 )
 
+func fillDefaultsForJaeger(span *model.Span) {
+	if span.OperationName == "" {
```

Comment:
[Jaeger author thinks](https://github.com/jaegertracing/jaeger/pull/3922) that having an empty span name is a valid edge cause and it shouldn't really trouble the span ingestion.

May be we should think of removing the constraint [enforced in the db](https://github.com/timescale/promscale_extension/blob/e6aaa0f3292f48caad326ff0de2ed1467f17d2d7/migration/incremental/012-tracing.sql#L99)?

Reply:
I am ok removing the constraint. You want to open a PR against the extension?

Reply:
@cevian FYI https://github.com/timescale/promscale_extension/pull/520